### PR TITLE
Publish keys to key server batching

### PR DESCRIPTION
**Pull request by jendakol**
Adjusted keys batching to follow rules about overlapping and alignment (required by Key server).

I screwed commit messages, ignore it. Will be squash anyway.

### CODE REVIEW

File: internal/functions/efgs/configuration.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ type publishConfig struct {
 	KeyServer          *utils.KeyServerConfig
 	Client             *http.Client
 	MaxBatchSize       int `env:"MAX_UPLOAD_KEYS,default=30"`
+	MaxKeyAge          int `env:"MAX_KEY_AGE,default=15"`
+	MaxOverlappingKeys int `env:"MAX_SAME_DAY_KEYS,default=15"`
```
Comment (marekaf) [inline internal/functions/efgs/configuration.go]: Can we keep the naming 1:1 with upstream? :D
https://github.com/google/exposure-notifications-server/blob/b3ac17696bd9c344fa9b4ba7eaac150a3d886d68/internal/generate/config.go#L44
https://github.com/google/exposure-notifications-server/blob/b3ac17696bd9c344fa9b4ba7eaac150a3d886d68/internal/generate/config.go#L45

File: internal/functions/efgs/keys-operations.go [hunk 2]
```diff
@@ -149,3 +156,83 @@ func signBatch(ctx context.Context, efgsEnv efgsutils.Environment, diagnosisKey
...
+
+	addNewChunk := func(key ExpKey) {
+		batches = append(batches, ExpKeyBatch{key})
+	}
+
+	// Determines whether the chunk is suitable for the key.
+	// 1) It must not have too much keys that overlap with the given one.
+	// 2) It must not have any unaligned keys.
+	isSuitable := func(chunk ExpKeyBatch, key ExpKey) bool {
+		if len(chunk) == batchSize {
```
Comment (marekaf) [inline internal/functions/efgs/keys-operations.go]: `>=` just to be sure? :D

File: internal/functions/efgs/keys-operations_test.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+			if tt.wantMatchResult && !reflect.DeepEqual(gotChunks, tt.wantChunks) {
+				t.Errorf("splitKeys() gotChunks = %+v, want %+v", gotChunks, tt.wantChunks)
+			}
+		})
+	}
+}
+
+func loadTestingKeys(suffix string) (efgsDownloadedKeys []ExpKey) {
+	bytes, err := ioutil.ReadFile(fmt.Sprintf("../../../testdata/efgs-downloaded-keys-%v.json", suffix))
```
Comment (marekaf) [inline internal/functions/efgs/keys-operations_test.go]: I'd move the `testdata` folder to `test/data` as per https://github.com/golang-standards/project-layout#test
I have some `*.json` req/resp samples in `example` folder but I'd move them to the new `test/data` too :)

File: internal/functions/efgs/keys-operations_test.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	}
+
+	if err = json.Unmarshal(bytes, &efgsDownloadedKeys); err != nil {
+		panic(err)
+	}
+
+	return efgsDownloadedKeys
+}
+
+// This is more-or-less copied from https://github.com/google/exposure-notifications-server/blob/release-0.9/internal/publish/model/exposure_model.go#L418
```
Comment (marekaf) [inline internal/functions/efgs/keys-operations_test.go:183]: should we ask them to move it from `internal` to `api` and reuse it as a lib?

**Comment by marekaf** [comment]
lgtm!

File: internal/functions/efgs/keys-operations_test.go (hunk 4 above)
Comment (jendakol) [reply internal/functions/efgs/keys-operations_test.go:183]: The original code is much more complex but yeah, we can try. However I think we're the only ones who would benefit from that.

File: internal/functions/efgs/keys-operations.go (hunk 2 above)
Comment (jendakol) [reply internal/functions/efgs/keys-operations.go]: All right :smile:

**Pull request merged**